core/src/test/java/io/parsingdata/metal/token/ParameterizedUntilTest.java

```diff
@@ -46,21 +47,21 @@
     @Parameters(name = "{0} ({4})")
     public static Collection<Object[]> data() {
         return Arrays.asList(new Object[][] {
-            { "[a,b,c,a,b,c] i=0,s=1,m=6 ab",     untilToken(0, 1, 6, con('c'), con("ab")),     stream("abcabc", US_ASCII), enc(), true },
```

Comment:
I'm not entirely sure if it's a good idea to truncate these tests in this way... They may currently parse in the same way as with the longer stream, but the reason for example that "abcabc" was provided for a parse that runs until(c) is to make sure the implementation stops at the _first_ match. So if you remove a second instance of "c", then the test no longer validates this.

The same is probably true for some of the other tests below. I think we should consider to maybe separate these tests for the standard parameterized implementation, or maybe make that implementation configurable, or something else...

Reply:
Yes, I agree about the missing validation that it stops at the first match. I don't think we need a to separate these tests though. We could include an additional def after the until, such that we parse the whole stream and still test if we stop at the first match. This was how I originally adjusted it, but then decided that this would make the tests too complex and I wanted to keep it simple.

Shall I change them back and then add a seq(until..., def('ABC')) around it?

Reply:
Either that or make the added check optional in `ParameterizedParse`. It could be standard with the default constructor, but with an optional constructor that disables that check.

I don't feel strongly about either solution, so if you do, then do it :) Otherwise we can talk about it next week.

Reply:
Ok, I have reated overload methods which introduce an argument for the expected tail of the stream, which wraps the until token and the expected tail in a seq.

Reply:
Looks good!